bittensor/_subtensor/subtensor_impl.py

```diff
@@ -510,62 +510,64 @@ def register (
                     
                 # pow successful, proceed to submit pow to chain for registration
                 else:
-                    #check if pow result is still valid
-                    if pow_result['block_number'] < self.get_current_block() - 3:
-                        bittensor.__console__.print( "[red]POW is stale.[/red]" )
-                        continue
-                    
-                    with self.substrate as substrate:
-                        # create extrinsic call
-                        call = substrate.compose_call( 
-                            call_module='SubtensorModule',  
-                            call_function='register', 
-                            call_params={ 
-                                'block_number': pow_result['block_number'], 
-                                'nonce': pow_result['nonce'], 
-                                'work': bittensor.utils.hex_bytes_to_u8_list( pow_result['work'] ), 
-                                'hotkey': wallet.hotkey.ss58_address, 
-                                'coldkey': wallet.coldkeypub.ss58_address
-                            } 
-                        )
-                        extrinsic = substrate.create_signed_extrinsic( call = call, keypair = wallet.hotkey )
-                        response = substrate.submit_extrinsic( extrinsic, wait_for_inclusion=wait_for_inclusion, wait_for_finalization=wait_for_finalization )
-                        
-                        # We only wait here if we expect finalization.
-                        if not wait_for_finalization and not wait_for_inclusion:
-                            bittensor.__console__.print(":white_heavy_check_mark: [green]Sent[/green]")
-                            return True
-                        
-                        # process if registration successful, try again if pow is still valid
-                        response.process_events()
-                        if not response.is_success:
-                            if 'key is already registered' in response.error_message:
-                                # Error meant that the key is already registered.
-                                bittensor.__console__.print(":white_heavy_check_mark: [green]Already Registered[/green]")
-                                return True
-
-                            bittensor.__console__.print(":cross_mark: [red]Failed[/red]: error:{}".format(response.error_message))
-                            time.sleep(1)
-                        
-                        # Successful registration, final check for neuron and pubkey
+                    with bittensor.__console__.status(":satellite: Submitting POW..."):
+                        # check if pow result is still valid
+                        while pow_result['block_number'] >= self.get_current_block() - 3:
```

Comment:
I can see this change of condition is correct.

Just a comment around this kind of changes:
- In terms of maintain the code it would be helpful to have the check of this condition encapsulated in a single function and also provide a unit test with a data provider with the set of inputs that lead us to have true/false. That would lead us to have a change of that test along with this condition change that would be helpful in terms of maintainability and also to read PRs around.

Reply:
extracted and wrote up some unit tests